core/autoscaling.go

```diff
@@ -334,6 +335,40 @@ func (a *autoScalingGroup) replaceOnDemandInstanceWithSpot(
 	return a.detachAndTerminateOnDemandInstance(odInst.InstanceId)
 }
 
+func (a *autoScalingGroup) getInstanceTypeByTagInASG() (string, error) {
```

Comment:
I don't understand what this function does and why we have it.

I think the tag is already available in the Tags attribute of each group, so we don't need to query it again.